projects/hathor-wallet-headless/api-docs-ci.md

```diff
@@ -0,0 +1,63 @@
+# Summary
+This feature automates the process of uploading the updated API Docs for the Headless Wallet, while giving the developer a preview of the contents, error checking and linting.
+
+A script is made available for the developer to make all the necessary checks locally and a GitHub Workflow is implemented to:
+- Validate and lint the `api-docs.js` file on every PR to the `dev` branch
+- Deploy the updated docs on every new successful release
+
+# Motivation
+Currently this upload is made manually by a team member, extracting data from the `src/api-docs.js` file and processing it to finally upload its contents to the [Headless Hathor Wallet API](https://wallet-headless.docs.hathor.network/) site.
+
+# Guide-level explanation
+### JS to JSON conversion
+The platform that manages the OpenAPI Documentation static website, [Redoc](https://github.com/Redocly/redoc), only interacts with `.json` files. So we need to convert our `src/api-docs.js` file into a `.json` one whenever we want to use this platform.
+
+A script should be built to make this conversion on the `scripts` folder, outputting the results to the `tmp/` folder. All following steps will be executed on this converted file. Some specific conditions to execute:
+- Remove the `securitySchemes` from the `components` property
+- Ensure the `default` property normally exported by the javascript `.toJSON()` is not present on the output json
+
+This script will be referred to as `scripts/convert-docs.js` from now on.
+
+### Linting and Preview
+A developer should be able to run a `npm run docs` command with three outcomes:
+- A validation check should be executed on the generated docs, alerting any errors found
+- A lint should be executed on the generated docs, displaying its analysis results
+- A local server should be fired, allowing the developer to visualize the results of the doc changes
+
+This would be an automated version of manually executing the following commands:
+```sh
+# Validation and conversion
+node ./scripts/convert-docs.js
+
+# Linting
+npx @redocly/cli lint tmp/api-docs.json
+
+# Local server for validating
+npx @redocly/cli preview-docs tmp/api-docs.json
+xdg-open http://127.0.0.1:8080
+```
+
+> *Note:* The linter throws some errors on our current documentation. The first implementation of this script will certainly require a refactoring PR.
+
+### PR Validation
+A GitHub Workflow, configured to run on every PR merging to `dev`, should run the conversion script and linter to check for errors on the generated documentation. This should be part of the CI process.
```

Comment:
Why only on PRs merging to `dev`? What if someone creates stacked PRs?

Reply:
Our current default is to only run our workflows on PRs merging to `dev` ( or, in cases where this branch doesn't exist, on `master` / `main` ).
If it becomes really necessary to run this on a separate branch, it would always be possible to add the branch name on the workflow `yml` file on that specific branch / PR.

What do you think?

Reply:
I think the problem with running tests/linters only for PRs based on `dev`/`master` is that PRs based on another feature branch wouldn't get their tests run.

This could lead us to introducing bugs when merging these PRs on top of one another, or at least taking longer to give some feedback for the developer about problems in their code.

For instance, I got one from our board to show you this is not so rare to happen: https://github.com/HathorNetwork/hathor-core/pull/645

This PR only had tests run on it because hathor-core is currently configured to run on any PR regardless of the base branch: https://github.com/HathorNetwork/hathor-core/blob/master/.github/workflows/main.yml#L10

Reply:
While I think this would be less necessary for a documentation feature, I think the overhead is very small ( the validation proposed is very fast and light on resources ) and the benefit is clear on the CI.

Removed the branch restriction on [a6d072b](https://github.com/HathorNetwork/rfcs/pull/56/commits/a6d072beae15f7f4700f450c3b9d7daf2b0b6077)